workflow/lib/preprocess/file_utils.py

```diff
@@ -1,28 +1,34 @@
-def get_sample_fps(samples_df, well=None, tile=None, cycle=None):
+import pandas as pd
+
+
+def get_sample_fps(
+    samples_df: pd.DataFrame, well: str = None, tile: int = None, cycle: int = None
+) -> list[str]:
     """
```

Comment:
I really like this approach. I have struggled heavily with wildcard usage downstream of preprocessing. Let's keep this up. Will be really useful for merge/hash (step 3)